librz/util/str.c

```diff
@@ -693,6 +694,96 @@ RZ_API const char *rz_sub_str_rchr(const char *str, int start, int end, char chr
 	return str[start] == chr ? str + start : NULL;
 }
 
+/**
+ * \brief Checks if the given character string is a two byte UTF-8 character.
+ *
+ * \param c The character string to test.
+ * \return bool True if the character string is a two byte UTF-8 character. False otherwise.
+ */
+RZ_API bool rz_str_is2utf8(RZ_NULLABLE const char *c) {
+	if (!c[0] || !c[1]) {
+		return false;
+	}
+	return ((c[0] & 0xe0) == 0xc0) && ((c[1] & 0xc0) == 0x80);
+}
+
+/**
+ * \brief Checks if the given character string is a three byte UTF-8 character.
+ *
+ * \param c The character string to test.
+ * \return bool True if the character string is a three byte UTF-8 character. False otherwise.
+ */
+RZ_API bool rz_str_is3utf8(RZ_NULLABLE const char *c) {
+	if (!c[0] || !c[1] || !c[2]) {
+		return false;
+	}
+	return ((c[0] & 0xf0) == 0xe0) && ((c[1] & 0xc0) == 0x80) && ((c[2] & 0xc0) == 0x80);
+}
+
+/**
+ * \brief Checks if the given character string is a four byte UTF-8 character.
+ *
+ * \param c The character string to test.
+ * \return bool True if the character string is a four byte UTF-8 character. False otherwise.
+ */
+RZ_API bool rz_str_is4utf8(RZ_NULLABLE const char *c) {
+	if (!c[0] || !c[1] || !c[2] || !c[3]) {
```

Comment:
i c is nullable then when null this will crash

Reply:
Yes, I suggest to mark them as `RZ_NONNULL` and add an assert